devolving_music/lib/song_scores.py

```diff
@@ -29,18 +31,35 @@ def get_scores(self) -> "dict[int, ScoreSuite]":
 
         return self.song_score_dict
 
+    def get_scores_list(self) -> "list[ScoreSuite]":
+        # calculates scores for all submissions using current comparisons
+        # returns list of all score suite objects
+        return list(self.get_scores().values())
+
     def get_compare_submission_random(self, submission_id) -> SongSubmission:
         key_list = list(self.song_score_dict.keys())
         key_list.remove(submission_id)
-        # once you have a critical number of comparisons then pull from quality
-        # list
-        return self.song_score_dict.get(random.choice(key_list)).song_submission
+        return self.song_score_dict.get(
+            random.choice(key_list)).song_submission
+
+    def get_compare_submission_closest(self, score_suite_obj: "ScoreSuite", information_threshold : int, update_scores = False):
+        # information threshold is the threshhold at which we start to consider score_suites as being able to be close to score_suite_obj
+        # score_suites below this information threshhold are considered to be infinitely far away from score_suite_obj
+        if(update_scores):
+            self.get_scores()
+        score_dict = copy.deepcopy(self.song_score_dict)
+        target_id = score_suite_obj.song_submission.id
+        score_dict.pop(target_id)
+        score_list=list(score_dict.values())
+        closest_songs=SongScores.get_distance_sort(score_suite_obj, score_list, information_threshold)
+        return closest_songs
```

Comment:
I still don't like the pattern of creating a variable for the sole purpose of immediately returning it. If the variable name tells you something about what it is, the function name should give the same information anyway.

Reply:
I still don't see why more information is bad. I personally like it. I think it makes things a lot easier to see than returning SongScores.get_distance_sort(score_suite_obj, score_list, information_threshold)